backend/substrapp/tasks/tasks.py

```diff
@@ -797,9 +805,40 @@ def _do_task(client, subtuple_directory, tuple_type, subtuple, compute_plan_id,
     with open(path.join(pred_path, 'perf.json'), 'r') as perf_file:
         perf = json.load(perf_file)
     result['global_perf'] = perf['all']
+
+    tag = subtuple.get("tag")
+    if tag and TAG_VALUE_FOR_TRANSFER_BUCKET in tag:
+        transfer_to_bucket(subtuple["key"], [pred_path, model_path])
+
     return result
 
 
+def transfer_to_bucket(tuple_key, paths):
+    if not ACCESS_KEY or not SECRET_KEY or not BUCKET_NAME:
+        logger.info(f'unset global env for bucket transter: {ACCESS_KEY} {SECRET_KEY} {BUCKET_NAME}')
+        return
+
+    tar_file = f'{tuple_key}.tar.gz'
+    tar_path = path.join('/tmp', tar_file)
+
+    with tempfile.TemporaryDirectory(prefix='/tmp/') as tmpdir:
+        tar_path = path.join(tmpdir, f'{tuple_key}.tar.gz')
+        try:
+            with tarfile.open(tar_path, 'x:gz') as tar:
+                for dir_path in paths:
+                    tar.add(dir_path)
+        except tarfile.TarError as e:
+            logger.warning(e)
```

Comment:
you should return here as in case of tar errors, you don't want to continue the transfer.